### ISSUE TITLE
[IE10]: SCRIPT5022: InvalidStateError webmakerLogin.min.js, line 2 character 334

### ISSUE THREAD

**Issue by humphd** [mozilla/webmaker-login-ux#169]
I'm testing bramble.mofostaging.net, which is using `webmakerLogin.min.js` (same as in Thimble), and in IE10 it throws doing this:

``` js
   for(var key in options.headers)
      xhr.setRequestHeader(key,options.headers[key]) // <------ here
```

The code is here https://github.com/mozilla/webmaker-login-ux/blob/e6d675f6ae3b14bc76b8f725de0f17155aa6269a/dist/ngWebmakerLogin.js#L213-L214.  This is a bit of a showstopper.  Has this been tested in IE for other apps?

cc @Pomax

**Comment by Pomax** [comment]
`for(... in ...)` needs to be refactored out of existence, it's from a JavaScript before there was `Object.keys`, and extremely brittle. With that said, what's the error the F12 dev tools for IE gives you? (is the key/value pair correct?)

(is [this stackoverflow question](http://stackoverflow.com/questions/22098259/access-denied-in-ie-10-and-11-when-ajax-target-is-localhost) applicable here?)

**Comment by humphd** [comment]
No, this isn't localhost.  I wonder if the issue is this: http://www.telerik.com/forums/custom-authorization-header-for-upload#FRi3NUFssUe8V3kzirsLoQ, and the readystatechange is happening twice, and the second time the headers get set it fails.

**Comment by Pomax** [comment]
This is caused by the browser-request package, which is probably the worst code I've read in a while... it's all over the place O_o

We _need_ to replace it with something like `superagent` instead, because it's doing all kinds of wrong things (IE is simply hitting a case that accidentally gets ignored by other browsers, involving xhr status 0) @jbuck @cadecairos

**Comment by humphd** [comment]
https://github.com/iriscouch/browser-request/pull/40 looks like something to explore here
